From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martijn Muijsers <martijnmuijsers@live.nl>
Date: Mon, 10 Jul 2023 22:05:08 +0200
Subject: [PATCH] Material fixes - Add missing values to hasGravity

Fixes the unit test for `Material.hasGravity`, as it was missing support for some falling blocks:
* Blocks that were an instance of `FallingBlock` but not `Fallable`
* Blocks that were an instance of `ScaffoldingBlock`

This makes the unit test work for the missing `Material` values added in the "Material fixes - Add missing values to hasGravity" patch in the `fiddle-api` module.

License: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)
Fiddle - https://fiddlemc.org

diff --git a/src/test/java/org/bukkit/PerMaterialTest.java b/src/test/java/org/bukkit/PerMaterialTest.java
index 9dab4e713b9bc628872867c218fe0ebeca0b958f..d2f7d3d2a690a31c6dc926fc9e5f7b51bdbd9aab 100644
--- a/src/test/java/org/bukkit/PerMaterialTest.java
+++ b/src/test/java/org/bukkit/PerMaterialTest.java
@@ -165,7 +165,10 @@ public class PerMaterialTest extends AbstractTestingBase {
     @EnumSource(value = Material.class, names = "LEGACY_.*", mode = EnumSource.Mode.MATCH_NONE)
     public void hasGravity(Material material) {
         if (material.isBlock()) {
-            assertThat(material.hasGravity(), is(CraftMagicNumbers.getBlock(material) instanceof FallingBlock));
+            // Fiddle start - add missing Material values with gravity
+            var block = CraftMagicNumbers.getBlock(material);
+            assertThat(material.hasGravity(), is(block instanceof net.minecraft.world.level.block.Fallable || block instanceof net.minecraft.world.level.block.ScaffoldingBlock));
+            // Fiddle end - add missing Material values with gravity
         } else {
             assertFalse(material.hasGravity());
         }
